src/Silverback.Integration/Messaging/Diagnostics/EmptyDiagnosticObserver.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) 2018-2019 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+
+namespace Silverback.Messaging.Diagnostics
+{
+    internal class EmptyDiagnosticObserver : IObserver<KeyValuePair<string, object>>

```

Comment:
Those things shouldn't be necessary anymore, if propagation is supported without dummy registration.
Be I will create an example how to use DiagnosticsSource with silverback (the same you can do it with framework internals)